backend/handler/user.go

```diff
@@ -220,6 +220,48 @@ func (h *UserHandler) Me(c echo.Context) error {
 	return c.JSON(http.StatusOK, map[string]string{"id": sessionToken.Subject()})
 }
 
+func (h *UserHandler) Delete(c echo.Context) error {
+	sessionToken, ok := c.Get("session").(jwt.Token)
+	if !ok {
+		return errors.New("missing or malformed jwt")
+	}
+
+	userId, err := uuid.FromString(sessionToken.Subject())
+	if err != nil {
+		return fmt.Errorf("failed to parse subject as uuid: %w", err)
+	}
+
+	return h.persister.Transaction(func(tx *pop.Connection) error {
+		user, err := h.persister.GetUserPersisterWithConnection(tx).Get(userId)
+		if err != nil {
+			return fmt.Errorf("failed to get user: %w", err)
+		}
+
+		if user == nil {
+			return fmt.Errorf("unknown user")
+		}
+
+		err = h.auditLogger.Create(c, models.AuditLogUserDeleted, user, nil)
```

Comment:
Shouldn't we first delete the use, then write to the audit log?

Reply:
Yes makes sense, I changed it. But I think the `AuditLogger.Create()` method should be able to work on a transaction, because currently we cannot guarantee that the audit log will be written.